spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :request do
+  describe 'GET #index' do
+    it 'renders a successful response' do
+      get user_posts_path(2)
+      expect(response).to be_successful
+    end
+
+    it 'renders the index template' do
+      get user_posts_path(5)
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes correct placeholder text in the response body' do
+      get user_posts_path(1)
+      expect(response.body).to include('<h1>Here are the posts for user:</h1>')
+    end
+  end
+
+  describe 'GET #show' do
+    it 'renders a successful response' do
+      get user_post_url(1, 3)
+      expect(response).to be_successful
+    end
+
+    it 'renders the correct template' do
+      get user_post_url(1, 3)
+      expect(response).to render_template('show')

```

Comment:
- [ ] [OPTIONAL] I suggest using the same syntax in your code so that when debugging or reading the code you can easily skim through (`:show` and `'show'`)